### PR TITLE
fix(cli): allow multiple occurrences of the same name in user properties

### PR DESCRIPTION
**Pull request by Red-Asuka**
### PR Checklist

If you have any questions, you can refer to the [Contributing Guide](https://github.com/emqx/MQTTX/blob/main/.github/CONTRIBUTING.md)

#### What is the current behavior?

Please describe the current behavior and link to a relevant issue.

#### Issue Number

Example: \#123

#### What is the new behavior?

image

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Specific Instructions

Are there any specific instructions or things that should be known prior to review?

### CODE REVIEW

File: cli/src/utils/parse.ts [hunk 1]
```diff
@@ -34,10 +34,23 @@ const parseMQTTVersion = (value: string) => {
...
 }
 
-const parseUserProperties = (value: string, previous: Record<string, unknown> | undefined) => {
+const parseUserProperties = (value: string, previous?: Record<string, string | string[]>) => {
   const [key, val] = value.split(': ')
   if (key && val) {
-    return previous ? { ...previous, [key]: val } : { [key]: val }
+    if (!previous) {
+      return { [key]: val }
+    } else {
```
Comment (ysfscream) [inline cli/src/utils/parse.ts:42]: `else` is needless after the `return` statement.

```
if (!previous) return { [key]: value }
if (xxx) {
    xxxx
}

File: cli/src/utils/parse.ts [hunk 2]
```diff
@@ -34,10 +34,23 @@ const parseMQTTVersion = (value: string) => {
...
+      if (previous[key]) {
+        if (Array.isArray(previous[key])) {
+          ;(previous[key] as string[]).push(val)
```
Comment (ysfscream) [inline cli/src/utils/parse.ts:45]: Please declare an internal variable to represent `previous[key]`, For example:

```js
const objValues = previous[key] as string[]
```

Looks cumbersome, but more clear type

File: cli/src/utils/parse.ts (hunk 1 above)
Comment (Red-Asuka) [reply cli/src/utils/parse.ts:42]: I know that `else` is unnecessary, but I think that in some cases `else` can make the code clearer. This is the code without `else`, what do you think?

```ts
    if (!previous) return { [key]: val }
    if (previous[key]) {
      if (Array.isArray(previous[key])) {
        ;(previous[key] as string[]).push(val)
      } else {
        previous[key] = [previous[key] as string, val]
      }
      return previous
    } else {
      return { ...previous, [key]: val }
    }
```

Comment (Red-Asuka) [reply cli/src/utils/parse.ts:42]: The final code:

```ts
    if (!previous) return { [key]: val }
    if (previous[key]) {
      if (Array.isArray(previous[key])) {
        const currentVal = previous[key] as string[]
        currentVal.push(val)
        return { ...previous, [key]: currentVal }
      }
      previous[key] = [previous[key] as string, val]
      return previous
    }
    return { ...previous, [key]: val }
```

Comment (Red-Asuka) [reply cli/src/utils/parse.ts:42]: Or
```ts
    if (!previous) return { [key]: val }
    if (!previous[key]) return { ...previous, [key]: val }
    if (Array.isArray(previous[key])) {
      const currentVal = previous[key] as string[]
      currentVal.push(val)
      return { ...previous, [key]: currentVal }
    }
    previous[key] = [previous[key] as string, val]
    return previous
```

Comment (ysfscream) [reply cli/src/utils/parse.ts:42]: `else` needs continuity, and `return` means execution is over. It doesn't say here that else doesn't need to be used, I think this code is just fine

```js
if (!previous) return { [key]: val }
if (previous[key]) {
  if (Array.isArray(previous[key])) {
	const currentVal = previous[key] as string[]
     currentVal.push(val)
  } else {
  	const currentVal = previous[key] as string
    previous[key] = [currentVal, val]
  }
  return previous
}
 return { ...previous, [key]: val }
```

**Review by ysfscream — APPROVED** [review]

File: cli/src/utils/parse.ts (hunk 1 above)
Comment (ysfscream) [reply cli/src/utils/parse.ts:42]: Anyway, I will merge first. But the principle of my judgment is that if there is a return in the if, there will be no need for else; if not, then there is a need for else; this proposal should be in the rules of `Eslint` also had, personal preference, code readability is our goal.

Comment (ysfscream) [reply cli/src/utils/parse.ts:42]: It's a nice job.

**Pull request merged**